### PR TITLE
Add ability to post to a user's account

### PR DESCRIPTION
**Pull request by arianna-h-h**

### CODE REVIEW

File: gistUpload.js [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
   const post = { [gist]: { content: contents } };
-  return axios.post(
-    API_URL,
-    { files: post },
-  )
+  return axios({
+    method: 'post',
+    url: API_URL,
+    auth: {
+      user: 'arianna-h-h',
```
Comment (snmgian) [inline gistUpload.js]: Shouldn't this come from `process.env` ?

**Review by snmgian — CHANGES_REQUESTED** [review]

File: gistUpload.js (hunk 1 above)
Comment (arianna-h-h) [reply gistUpload.js]: done

File: gistUpload.js [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
 const BAR_LENGTH = 30;
 const HANG_PERCENT = 0.25;
 
+
 /** This is a description of the uploadFile function
  * @param {string} contents - Contents of file from readFile
  * @param {string} gist - Name of gist supplied by user
  */
 function uploadFile(contents, gist) {
   const post = { [gist]: { content: contents } };
```
Comment (jorions) [inline gistUpload.js]: Try to use variable names/definitions to make your code more concise. To clean this up you could say
` function uploadFile(content, gist) {`
`const files = { files: { [gist]: { content } } }`
`return axios.post(API_URL, files)`

**Review by jorions — CHANGES_REQUESTED** [review]

File: gistUpload.js (hunk 2 above)
Comment (arianna-h-h) [reply gistUpload.js]: I am able to cut line 17 out, but I have to use the full 
````
return axios(method..., url, auth..., data..)
````
in order to correctly authenticate to github's API.

File: gistUpload.js [hunk 3]
```diff
@@ -14,15 +14,14 @@ const HANG_PERCENT = 0.25;
...
-  const post = { [gist]: { content: contents } };
   return axios({
     method: 'post',
     url: API_URL,
     auth: {
       user: process.env.USER,
       password: process.env.SECRET_TOKEN,
     },
-    data: { files: post },
+    data: { files: { [gist]: { content: contents } } },
```
Comment (jorions) [inline gistUpload.js]: rename `contents` argument to `content` so that you can say `data: { files: { [gist]: { content } } }`

**Review by jorions — CHANGES_REQUESTED** [review]

**Review by snmgian — APPROVED** [review]

**Pull request merged**